src/ray/common/asio/instrumented_io_context.h

```diff
@@ -28,7 +28,11 @@ class instrumented_io_context : public boost::asio::io_context {
  public:
   /// Initializes the global stats struct after calling the base contructor.
   /// TODO(ekl) allow taking an externally defined event tracker.
-  instrumented_io_context() : event_stats_(std::make_shared<EventTracker>()) {}
+  ///
+  /// \param enable_lag_probe If true, and if related Ray configs are set, schedule a
+  /// probe to measure the event loop lag. After a probe is done, it schedules another one
+  /// so a io_context.run() call will never return. Should only be set to false for tests.
+  explicit instrumented_io_context(bool enable_lag_probe = true);
```

Comment:
Let's do opt-in first instead of opt-out? To start with, let's only enable for GCS: enabling for all core workers will generate too many metrics for a large cluster

Reply:
done. now it's opt-in, and only gcs server opts in.